crates/bevy_remote/src/lib.rs

```diff
@@ -0,0 +1,854 @@
+    /// any associated methods.
+    fn empty() -> Self {
+        Self {
+            address: DEFAULT_ADDR,
+            port: DEFAULT_PORT,
+            methods: RwLock::new(vec![]),
+        }
+    }
+
+    /// Set the IP address that the server will use.
+    #[must_use]
+    pub fn with_address(mut self, address: impl Into<IpAddr>) -> Self {
+        self.address = address.into();
+        self
+    }
+
+    /// Set the remote port that the server will listen on.
+    #[must_use]
+    pub fn with_port(mut self, port: u16) -> Self {
+        self.port = port;
+        self
+    }
+
+    /// Add a remote method to the plugin using the given `name` and `handler`.
+    #[must_use]
+    pub fn with_method<M>(
+        mut self,
+        name: String,
+        handler: impl IntoSystem<Option<Value>, BrpResult, M>,
+    ) -> Self {
+        self.methods
+            .get_mut()
+            .unwrap()
+            .push((name, Box::new(IntoSystem::into_system(handler))));
+        self
+    }
+}
+
+impl Default for RemotePlugin {
+    fn default() -> Self {
+        Self::empty()
+            .with_method(
+                "bevy/get".to_owned(),
+                builtin_methods::process_remote_get_request,
+            )
+            .with_method(
+                "bevy/query".to_owned(),
+                builtin_methods::process_remote_query_request,
+            )
+            .with_method(
+                "bevy/spawn".to_owned(),
+                builtin_methods::process_remote_spawn_request,
+            )
+            .with_method(
+                "bevy/insert".to_owned(),
+                builtin_methods::process_remote_insert_request,
+            )
+            .with_method(
+                "bevy/remove".to_owned(),
+                builtin_methods::process_remote_remove_request,
+            )
+            .with_method(
+                "bevy/destroy".to_owned(),
+                builtin_methods::process_remote_destroy_request,
+            )
+            .with_method(
+                "bevy/reparent".to_owned(),
+                builtin_methods::process_remote_reparent_request,
+            )
+            .with_method(
+                "bevy/list".to_owned(),
+                builtin_methods::process_remote_list_request,
+            )
+    }
+}
+
+impl Plugin for RemotePlugin {
+    fn build(&self, app: &mut App) {
+        let mut remote_methods = RemoteMethods::new();
+        let plugin_methods = &mut *self.methods.write().unwrap();
+        for (name, system) in plugin_methods.drain(..) {
+            remote_methods.insert(
+                name,
+                app.main_mut().world_mut().register_boxed_system(system),
+            );
+        }
+
+        app.insert_resource(HostAddress(self.address))
+            .insert_resource(HostPort(self.port))
+            .insert_resource(remote_methods)
+            .add_systems(Startup, start_server)
+            .add_systems(Update, process_remote_requests);
```

Comment:
Not in scope for this PR, but mixing "editor" logic with game logic will cause trouble in debugging, like during system stepping you won't be able to use BRP, because it's just a "game logic" system.
Is there a proposed solution for this?

Reply:
Probably to special-case out the systems that make the editor 'embassy' tools like this work. This discussion might be better suited to discord.